### PR TITLE
Make the HTML interpreter more approachable

### PR DESCRIPTION
**Pull request by CosmicHorrorDev**
This is a big bundle of changes that tries to make the code for the interpreter a decent bit more approachable (although it still has a long way to go in terms of larger refactors)

~~Draft since I'm going to go back through the history and clean things up before merging. There were several bugs that got fixed during the changes~~

This contains a myriad of various improvements (more exhaustive matching, logging more interesting things, reworking and adding a whole slew of constructors, etc.) all with the goal of making the `impl TokenSink for HtmlInterpreter` implementation more approachable

### CODE REVIEW

**Comment by CosmicHorrorDev** [comment]
@Valentin271 Not sure if you're interested in reviewing, but I would highly recommend switching to hide whitespace when looking through `interpreter/mod.rs` or else the diff is abysmal (not sure if that's a beta feature or not)

image

File: src/interpreter/html.rs [hunk 1]
```diff
@@ -248,3 +252,120 @@ pub enum Element {
...
+    Summary,
+    Table,
+    TableBody,
+    TableDataCell,
+    TableHead,
+    TableHeader,
+    TableRow,
+    Underline,
+    UnorderedList,
+}
```
Comment (Valentin271) [inline src/interpreter/html.rs:325]: This is an important and large part of the interpreter, I'd suggest moving it to another subfile.
`Element` could also benefit from moving to another subfile

**Review by Valentin271 — APPROVED** [review]
This is a good first step towards simplifying this.
In general this looks good, my only comment would be about trying to split types in more files, to have more concise files.

**Comment by CosmicHorrorDev** [comment]
Agreed. Split them both into their own files 👍

**Pull request merged**